oap-server/server-receiver-plugin/skywalking-trace-receiver-plugin/src/main/java/org/apache/skywalking/oap/server/receiver/trace/provider/TraceModuleProvider.java

```diff
@@ -75,8 +77,11 @@ public void prepare() throws ServiceNotProvidedException {
 
         uninstrumentedGatewaysConfig = new UninstrumentedGatewaysConfig(this);
 
+        traceSampleRateWatcher = new TraceSampleRateWatcher(String.valueOf(moduleConfig.getSampleRate()), this);
```

Comment:
I think should not change to String here, please move this into the `TraceSampleRateWatcher`.

Reply:
ok